<?xml version="1.0"?>
<robot name="atlas">
  <link name="base_link"/>
  <link name="ground"/>
  <link name="model">
    <visual>
      <origin xyz="-0.9 0.0 0.0" rpy="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://atlas2_description/meshes/mitsubishi_miev.dae"/>
      </geometry>
    </visual>
  </link>
  <joint name="base_link_to_model" type="fixed">
    <origin xyz="0.0 0.0 -0.4" rpy="0.0 0.0 1.57079632679"/>
    <parent link="base_link"/>
    <child link="model"/>
  </joint>
  <joint name="base_link_to_ground" type="fixed">
    <origin xyz="0.0 0.0 -0.4" rpy="0.0 0.0 0.0"/>
    <parent link="base_link"/>
    <child link="ground"/>
  </joint>
  <link name="top_right_camera">
    <inertial>
      <mass value="0.085"/>
      <inertia ixx="1e-06" ixy="0.0" ixz="0.0" iyy="1e-06" iyz="0.0" izz="1e-06"/>
    </inertial>
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
      <geometry>
        <box size="0.0754 0.029 0.035"/>
      </geometry>
      <material name="dark_grey"/>
    </visual>
    <collision>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
      <geometry>
        <box size="0.0754 0.029 0.035"/>
      </geometry>
    </collision>
  </link>
  <link name="top_right_camera_optical"/>
  <joint name="top_right_camera_optical_joint" type="fixed">
    <origin xyz="0.0 0.0 0.0" rpy="-1.570796 0.0 -1.570796"/>
    <parent link="top_right_camera"/>
    <child link="top_right_camera_optical"/>
  </joint>
  <gazebo reference="top_right_camera_optical"/>
  <joint name="base_link_to_top_right_camera" type="fixed">
    <origin xyz="-1.77280187607 -0.233933746815 1.25" rpy="-6.41765959721e-08 0.236651372034 -0.0570543483677"/>
    <parent link="base_link"/>
    <child link="top_right_camera"/>
  </joint>
  <link name="top_left_camera">
    <inertial>
      <mass value="0.085"/>
      <inertia ixx="1e-06" ixy="0.0" ixz="0.0" iyy="1e-06" iyz="0.0" izz="1e-06"/>
    </inertial>
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
      <geometry>
        <box size="0.0754 0.029 0.035"/>
      </geometry>
      <material name="dark_grey"/>
    </visual>
    <collision>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
      <geometry>
        <box size="0.0754 0.029 0.035"/>
      </geometry>
    </collision>
  </link>
  <link name="top_left_camera_optical"/>
  <joint name="top_left_camera_optical_joint" type="fixed">
    <origin xyz="0.0 0.0 0.0" rpy="-1.570796 0.0 -1.570796"/>
    <parent link="top_left_camera"/>
    <child link="top_left_camera_optical"/>
  </joint>
  <gazebo reference="top_left_camera_optical"/>
  <joint name="base_link_to_top_left_camera" type="fixed">
    <origin xyz="-1.73982644081 0.382407188416 1.25" rpy="4.58485844927e-05 0.226041151401 -0.0962249358456"/>
    <parent link="base_link"/>
    <child link="top_left_camera"/>
  </joint>
  <link name="frontal_laser_mount_link">
    <inertial>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
      <mass value="0.001"/>
      <inertia ixx="1e-06" ixy="0.0" ixz="0.0" iyy="1e-06" iyz="0.0" izz="1e-06"/>
    </inertial>
  </link>
  <joint name="frontal_laser_joint" type="fixed">
    <origin xyz="0.044 0.02325 0.0" rpy="0.0 0.0 0.0"/>
    <parent link="frontal_laser_mount_link"/>
    <child link="frontal_laser"/>
  </joint>
  <link name="frontal_laser">
    <inertial>
      <origin xyz="0.0026 -0.02325 -0.0" rpy="0.0 0.0 0.0"/>
      <mass value="1.0"/>
      <inertia ixx="0.00290035300653" ixy="0.0" ixz="0.0" iyy="0.00136918611899" iyz="0.0" izz="0.00297887297512"/>
    </inertial>
    <visual>
      <origin xyz="-0.044 -0.02325 -0.0" rpy="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://sick_ldmrs_description/meshes/sick_ldmrs.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="-0.044 -0.02325 -0.0" rpy="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://sick_ldmrs_description/meshes/sick_ldmrs.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="base_link_to_frontal_laser_mount_link" type="fixed">
    <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    <parent link="base_link"/>
    <child link="frontal_laser_mount_link"/>
  </joint>
  <link name="right_laser">
    <inertial>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
      <mass value="1.1"/>
      <inertia ixx="0.00307156543804" ixy="0.0" ixz="0.0" iyy="0.00312849041527" iyz="0.0" izz="0.00196432421427"/>
    </inertial>
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://lms1xx/meshes/sick-lms1xx.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0.0 0.0 -0.05" rpy="0.0 0.0 0.0"/>
      <geometry>
        <box size="0.105 0.1 0.1"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="right_laser"/>
  <joint name="base_link_to_right_laser" type="fixed">
    <origin xyz="-0.0500000007451 -0.699999988079 0.10000000149" rpy="7.91199037381e-09 5.07742774923e-10 -0.802577692221"/>
    <parent link="base_link"/>
    <child link="right_laser"/>
  </joint>
  <link name="left_laser">
    <inertial>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
      <mass value="1.1"/>
      <inertia ixx="0.00307156543804" ixy="0.0" ixz="0.0" iyy="0.00312849041527" iyz="0.0" izz="0.00196432421427"/>
    </inertial>
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://lms1xx/meshes/sick-lms1xx.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0.0 0.0 -0.05" rpy="0.0 0.0 0.0"/>
      <geometry>
        <box size="0.105 0.1 0.1"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="left_laser"/>
  <joint name="base_link_to_left_laser" type="fixed">
    <origin xyz="-0.0500000007451 0.699999988079 0.10000000149" rpy="-1.40600650738e-08 1.04046331119e-08 0.850917484309"/>
    <parent link="base_link"/>
    <child link="left_laser"/>
  </joint>
  <sensor name="top_right_camera" update_rate="20.0">
    <parent link="top_right_camera_optical"/>
    <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    <calibration_parent link="base_link"/>
    <calibration_child link="top_right_camera"/>
    <topic topic="/top_right_camera/image_color"/>
  </sensor>
  <sensor name="top_left_camera" update_rate="20.0">
    <parent link="top_left_camera_optical"/>
    <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    <calibration_parent link="base_link"/>
    <calibration_child link="top_left_camera"/>
    <topic topic="/top_right_camera/image_color"/>
  </sensor>
  <sensor name="left_laser" update_rate="20.0">
    <parent link="left_laser"/>
    <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    <calibration_parent link="base_link"/>
    <calibration_child link="left_laser"/>
    <topic topic="/left_laser/laserscan"/>
  </sensor>
  <sensor name="right_laser" update_rate="20.0">
    <parent link="right_laser"/>
    <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    <calibration_parent link="base_link"/>
    <calibration_child link="right_laser"/>
    <topic topic="/right_laser/laserscan"/>
  </sensor>
</robot>
